[READY] Working on basic windows(menu, game)

Transliteration opened the pull request:
Взял пример с той ссылки и теперь в меню:
1 по нажатию Enter переход в окно игры;
2 по нажатию Esc переход в Нарнию(выход из игрЫ).
В окне игры можно двигаться и по нажатию Esc переход в меню.
Оставил комменты в тех местах, которые считаю сильно косячными.
И добавлю здесь, что не знаю куда лучше сохранить засунуть переменную PATH.

Transliteration commented:
https://github.com/SFML/SFML/wiki/Tutorial:-Manage-different-Screens
Та самая ссылка

makometr commented on Source/screen_menu.hpp:
```diff
@@ -0,0 +1,91 @@
+	if (!font.loadFromFile(RESOURCES_PATH_PREFIX + "coopbl.ttf"))
+	{
+		std::cout << "Font didn't load!" << std::endl;
+		throw;
```
**да, ты прав:)** при компиляции руками наткнулся на ошибку необработанного исключения и завершения программы
Мало того, что нужно было написать обработчик исключения catch, так ещё и throw нужно по-другому использовать.
При броске без аргументов `throw;` он автоматически пробрасывается вверх по стеку вызовов (если я правильно помню), а сверху обработчика-то и нет=)
@Transliteration

makometr replied:
ага, увидел коммент

makometr commented:
Подправить свистопляску с заголовочными нужно

Denmey commented:
I think we should rename cScreen to ScreenBase, screenGame to ScreenGame and etc. Function Run in cScreen class should be renamed to run
Also we can put their source files in separate directory called Screen or Screens

Transliteration commented:
Renamed: function Run to run, class names cScreen to ScreenBase, screenGame to ScreenGame and etc.
Added separate folder **Screens** for Screen* classes.

makometr commented on Source/td.cpp:
```diff
@@ -23,22 +25,24 @@ int main()
+	ScreenGame s_game;
 	
+
+	// Не понял зачем здесь try-catch, необработанное исключение же в конструкторе
```
да, конечно, я промахнулся ночью

Denmey commented on CMakeLists.txt:
```diff
@@ -1,13 +1,14 @@
 project(Tetris)
    
 # Adding source files
-file(GLOB src Source/*.cpp)
```
Is "Source/*.cpp" didn't work for nested directory?

Denmey commented on Source/Screens/ScreenGame.hpp:
```diff
@@ -0,0 +1,14 @@
+private:
+	float movement_step;
+	float posx;
+	float posy;
```
What's an posx, posy and movement_step purpose?(Are posx and posy width and height? If they're a position then position of what they describe?) Rename them for better readability if you can.
Additionally change their names to camelCase

Denmey commented on Source/Screens/ScreenMenu.cpp:
```diff
@@ -0,0 +1,74 @@
+
+ScreenMenu::ScreenMenu() : playing(false) 
+{
+	if (!font.loadFromFile(RESOURCES_PATH_PREFIX + "coopbl.ttf"))
```
Put fonts in "Fonts" subdirectory

Denmey commented on Source/Screens/ScreenMenu.hpp:
```diff
@@ -0,0 +1,15 @@
+
+class ScreenMenu : public ScreenBase {
+private:
+	bool playing;
```
I think it's better to rename "playing" to "isPlaying". Name variables for state of something like this in future.
Should we add prefix m_ to all members to prevent naming collisions with members and functions like it's done in SFML?(m_isRepeated member and isRepeated() function, see [SFML/Graphics/Texture.hpp](https://github.com/SFML/SFML/blob/master/include/SFML/Graphics/Texture.hpp))

makometr replied:
There are so many prefixes in different libs but pointer **this** is one! 
If collision appears, **this** will solve this problem easily

Denmey commented on Source/td.cpp:
```diff
@@ -1,38 +1,49 @@
-  return 0;
-}
+	//Applications variables
+	std::vector<ScreenBase*> Screens;
```
Pure pointers are bad! Instead of them use unique_ptr, shared_ptr and weak_ptr.

makometr replied:
So, we can incapsulate this vector in individual class in the future

makometr commented on Source/td.cpp:
```diff
@@ -1,11 +1,13 @@
-	std::vector<cScreen*> Screens;
+	std::vector<ScreenBase*> Screens;
+
+	// Сомнительное резервирование памяти (зочем?)
```
Yes, I understand, there is no any performance improvment in this case. I hope, it will be interesting for u
![image](https://user-images.githubusercontent.com/21985069/43849139-61249e6e-9b3d-11e8-89e5-e952fcf4fd34.png)
![image](https://user-images.githubusercontent.com/21985069/43849181-7cee582e-9b3d-11e8-841e-a2d4e6f35fa8.png)

Transliteration commented:
Is "Source/*.cpp" didn't work for nested directory? No, it doesnt.
What's an posx, posy... Итс затычка для показа работы нескольких скринов, не вижу смысла что-то менять.
